### PR TITLE
Anonymous harvester factory

### PR DESCRIPTION
**Pull request by ajryan**
For issue #33 
Hopefully this matches what you were expecting, thanks!

### CODE REVIEW

File: StatePrinter.Tests/FieldHarvesters/AnonymousHarvesterTest.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        }
+
+        public void SpecializedClassHandler()
+        {
+            var sut = CreatePrinter();
+            var expected = @"new B()
+{
+ Age = ""Its age is 1""
+}";
+            var actual = sut.PrintObject(new B { Age = 1 });
```
Comment (kbilsted) [inline StatePrinter.Tests/FieldHarvesters/AnonymousHarvesterTest.cs]: maybe add another test instantiating an A instance and showing it is not harvested?

Comment (ajryan) [inline StatePrinter.Tests/FieldHarvesters/AnonymousHarvesterTest.cs]: Will do!

**Comment by ajryan** [comment]
Thanks for looking at this so quickly. Agreed about the messy commits, I'll rebase. And I'll add a bit to the [field harvesting](https://github.com/kbilsted/StatePrinter/blob/master/doc/HowToConfigure.md#15-field-harvesting) section of the docs.

File: StatePrinter.Tests/FieldHarvesters/AnonymousHarvesterTest.cs (hunk 1 above)
Comment (kbilsted) [inline StatePrinter.Tests/FieldHarvesters/AnonymousHarvesterTest.cs]: great

Comment (kbilsted) [inline StatePrinter.Tests/FieldHarvesters/AnonymousHarvesterTest.cs]: regarding the rebase,  i did spend time wondering and commenting on the files.. only to see them deleted after making the comments ;)

**Comment by ajryan** [comment]
Sorry for the mess, I am getting back into OSS contributing after being away from a while... I think this is clean now! Added a negative test and some docs. Thanks.

**Comment by ajryan** [comment]
woo thanks!

**Comment by kbilsted** [comment]
No problem @ajryan  - maybe you'll find some of the other issues fun to poke at. Or make up your own issues if you want ;)

**Pull request merged**